core/app/models/spree/store_channel.rb

```diff
@@ -0,0 +1,8 @@
+module Spree
+  class StoreChannel < ApplicationRecord
+    belongs_to :store, class_name: 'Spree::Store'
+    has_many :orders, class_name: 'Spree::Order', foreign_key: :channel_id
+
+    validates :name, presence: true
```

Comment:
also we need unique validation for name - there shouldn't be duplicate entries for the same store